graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    /**
+     * Sets the {@link GraphQL.Builder} consumer.
+     */
+    public GraphQLServiceBuilder configure(Iterable<Consumer<GraphQL.Builder>> configurers) {
+        configurerBuilder.addAll(requireNonNull(configurers, "configurers"));
+        return this;
+    }
+
+    /**
+     * Creates a {@link GraphQLService}.
+     */
+    public GraphQLService build() {
+        final TypeDefinitionRegistry registry = typeDefinitionRegistry(schemaFileBuilder.build());
+
+        final RuntimeWiring runtimeWiring = buildRuntimeWiring(runtimeWiringConfiguratorBuilder.build());
+        GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(registry, runtimeWiring);
+        final List<GraphQLTypeVisitor> typeVisitors = typeVisitorBuilder.build();
+        for (GraphQLTypeVisitor typeVisitor : typeVisitors) {
+            schema = SchemaTransformer.transformSchema(schema, typeVisitor);
+        }
+
+        GraphQL.Builder builder = GraphQL.newGraphQL(schema);
+        final List<Instrumentation> instrumentations = instrumentationBuilder.build();
+        if (!instrumentations.isEmpty()) {
+            builder = builder.instrumentation(new ChainedInstrumentation(instrumentations));
+        }
+
+        final List<Consumer<GraphQL.Builder>> configurers = configurerBuilder.build();
+        for (Consumer<GraphQL.Builder> configurer : configurers) {
+            configurer.accept(builder);
+        }
+        return new DefaultGraphQLService(builder.build(), dataLoaderRegistry);
+    }
+
+    private static TypeDefinitionRegistry typeDefinitionRegistry(List<File> schemaFiles) {
+        final TypeDefinitionRegistry registry = new TypeDefinitionRegistry();
+        final SchemaParser parser = new SchemaParser();
+        if (schemaFiles.isEmpty()) {
+            schemaFiles = defaultSchemaFiles();
+        }
+        if (schemaFiles.isEmpty()) {
+            throw new IllegalStateException("Not found schema file(s)");
+        }
+        schemaFiles.forEach(it -> registry.merge(parser.parse(it)));
+        return registry;
+    }
+
+    private static RuntimeWiring buildRuntimeWiring(
+            List<RuntimeWiringConfigurator> runtimeWiringConfigurators) {
+        final Builder runtimeWiringBuilder = RuntimeWiring.newRuntimeWiring();
+        runtimeWiringConfigurators.forEach(it -> it.configure(runtimeWiringBuilder));
+        return runtimeWiringBuilder.build();
+    }
+
+    private static List<File> defaultSchemaFiles() {
+        final ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        final ImmutableList.Builder<File> builder = ImmutableList.builder();
+        resourcePath(classLoader, "schema.graphqls").ifPresent(it -> builder.add(toFile(it)));
+        resourcePath(classLoader, "schema.graphql").ifPresent(it -> builder.add(toFile(it)));
+        return builder.build();
+    }
+
+    private static Optional<URL> resourcePath(ClassLoader classLoader, String resourcePath) {
```

Comment:
Let' use `@Nullable`.
```suggestion
    @Nullable
    private static URL resourcePath(ClassLoader classLoader, String resourcePath) {
```